src/CaptureClient/CaptureEventProcessor.cpp

```diff
@@ -363,9 +367,78 @@ void CaptureEventProcessorForListener::ProcessGpuQueueSubmission(
   }
 }
 
-void CaptureEventProcessorForListener::ProcessSystemMemoryUsage(
+void CaptureEventProcessorForListener::ProcessMemoryUsageEvent(
+    const orbit_grpc_protos::MemoryUsageEvent& memory_usage_event) {
+  if (memory_usage_event.has_system_memory_usage()) {
+    ExtractAndProcessSystemMemoryTrackingTimer(memory_usage_event.timestamp_ns(),
+                                               memory_usage_event.system_memory_usage());
+  }
+
+  if (memory_usage_event.has_cgroup_memory_usage() &&
+      memory_usage_event.has_process_memory_usage()) {
+    ExtractAndProcessCGroupAndProcessMemoryTrackingTimer(memory_usage_event.timestamp_ns(),
+                                                         memory_usage_event.cgroup_memory_usage(),
+                                                         memory_usage_event.process_memory_usage());
+  }
+}
+
+void CaptureEventProcessorForListener::ExtractAndProcessSystemMemoryTrackingTimer(
+    uint64_t synchronized_timestamp_ns,
     const orbit_grpc_protos::SystemMemoryUsage& system_memory_usage) {
-  capture_listener_->OnSystemMemoryUsage(system_memory_usage);
+  TimerInfo timer;
+  timer.set_type(TimerInfo::kSystemMemoryUsage);
+  timer.set_start(synchronized_timestamp_ns);
+  timer.set_end(synchronized_timestamp_ns);
```

Comment:
This supports even more that `TimerInfo` should just not be used here, this is a single point in time, not a time slice, which is what `TimerInfo` is supposed to represent.

Reply:
Make sense. I have added a bug for using dedicated structs for the memory tracks and also add a TODO here. Thanks for pointing out this!